operators/pkg/instanceCreation/creation.go

```diff
@@ -22,18 +22,78 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func CreateVirtualMachineInstance(name string, namespace string, template crownlabsv1alpha1.LabTemplate, instanceName string, secretName string) virtv1.VirtualMachineInstance {
-	vm := template.Spec.Vm
-	vm.Name = name + "-vmi"
-	vm.Namespace = namespace
-	vm.Labels = map[string]string{"name": name, "template-name": template.Name, "instance-name": instanceName}
+var terminationGracePeriod int64 = 30
 
-	for _, volume := range vm.Spec.Volumes {
-		if volume.Name == "cloudinitdisk" {
-			volume.CloudInitNoCloud.UserDataSecretRef = &corev1.LocalObjectReference{Name: secretName}
-		}
+func CreateVirtualMachineInstance(name string, namespace string, template crownlabsv1alpha2.Environment, instanceName string, secretName string) (*virtv1.VirtualMachineInstance, error) {
+	vm := virtv1.VirtualMachineInstance{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name + "-vmi",
+			Namespace: namespace,
+			Labels:    map[string]string{"name": name, "template-name": template.Name, "instance-name": instanceName},
+		},
+		Spec: virtv1.VirtualMachineInstanceSpec{
+			TerminationGracePeriodSeconds: &terminationGracePeriod,
+			Domain: virtv1.DomainSpec{
+				Resources: virtv1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						"cpu":    template.Resources.ReservedCPU,
+						"memory": template.Resources.Memory,
+					},
+					Limits: v1.ResourceList{
+						"cpu":    template.Resources.ReservedCPU,
+						"memory": template.Resources.Memory,
+					},
```

Comment:
This should take into account the cpunumber and not ReservedCPU.

In addition, you should add some extra room for the hypervisor as we are doing manually now.